docs/HACKING.md

```diff
@@ -1,104 +1,181 @@
-- Verify push/pull access to a container registry from where the cluster can pull image from.
-
-## User Workflows
-
-### I want to install an officially released version of WMCO on my OpenShift/OKD cluster
-* Instructions to install the community operator can be found in the [README](/README.md).
-  Customers with a Red Hat subscription should instead look at the [OpenShift docs.](https://docs.openshift.com/container-platform/latest/windows_containers/enabling-windows-container-workloads.html)
-
-### I want to try the unreleased WMCO
-* [Build the operator](#build)
-* [Deploy the operator directly](#deploying-the-operator-directly)
-
-### I want to try the unreleased WMCO, and deploy it using OLM
-* [Build the operator](#build)
-* [Deploy the operator through OLM](#deploying-the-operator-through-olm)
-
-### I want to try the unreleased WMCO, and deploy it mimicking the official deployment method
-* [Build the operator](#build)
-* [Generate bundle manifests](#generating-new-bundle-manifests)
-* [Build the bundle image](#creating-a-bundle-image)
-* [Build an operator index](#creating-a-new-operator-index)
-* [Create a CatalogSource using the index](https://docs.openshift.com/container-platform/latest/operators/admin/olm-managing-custom-catalogs.html#olm-creating-catalog-from-index_olm-managing-custom-catalogs)
-* [Follow the normal procedure to deploy an operator](https://docs.openshift.com/container-platform/latest/operators/admin/olm-adding-operators-to-cluster.html)
+
+This guide will walk you through the workflows required to run local e2e tests on the Windows Machine Config Operator
+(WMCO), and perform tasks like update the submodule versions, and deploy custom workloads on a cluster.
+
+## Cluster Configuration
+
+before running tests, your cluster must be properly configured with the correct networking and be running the proper
+version of OpenShift for use with the version of WMCO you are testing against. For example, when testing a 4.10 release
+of the WMCO, your cluster should be running a 4.10 version of OpenShift.
+Changes made against master should be tested against the latest CI version of OpenShift.
+
+If you already have a configured cluster, move on to [Build](#build)
+
+### Pre-cluster checklist
+
+- Download the correct OpenShift installer
+  - [OpenShift Installer Mirror](<https://mirror.openshift.com/pub/openshift-v4/clients/ocp/>)
+  - [OpenShift Nightly Builds](<https://amd64.ocp.releases.ci.openshift.org/#4-dev-preview>)
+- [Initialize git submodules](<https://github.blog/2016-02-01-working-with-submodules/>)
+- [Verify push/pull access to quay.io](<https://quay.io>)
+- Create an install-config.yaml
+  - [AWS](https://console.redhat.com/openshift/install/aws/installer-provisioned)
+  - [GCP](https://docs.openshift.com/container-platform/4.12/installing/installing_gcp/installing-gcp-default.html)
+  - [Azure](https://learn.microsoft.com/en-us/azure/openshift/tutorial-create-cluster)
```

Comment:
Please link to https://docs.openshift.com/container-platform/4.12/installing/installing_azure/installing-azure-default.html